tests/test.py

```diff
@@ -0,0 +1,100 @@
+import itertools
+import os
+import subprocess
+import sys
+
+if sys.version_info.major == 2:
+    from backports import tempfile
+else:
+    import tempfile
+
+import pytest
+
+HERE = os.path.realpath(os.path.dirname(__file__))
+PACKAGE_DIR = os.path.realpath(os.path.join(HERE, '..'))
+
+TOX_VERSIONS = [
+    ">=3.7,<3.8",
+    ">=3.8,<3.9",
+]
+
+TOX_TO_TOX_VENV_VERSIONS = {
+    # tox-venv 0.4.0 requires tox>=3.8.1
+    ">=3.7,<3.8": "<0.4.0",
+    ">=3.8,<3.9": ">=0.4.0",
+}
+
+CASES = {
+    "test-two-envs": {
+        "env": {},
+    },
+    "test-env-inheritance": {
+        "env": {},
+    },
+    "test-environment-variable": {
+        "env": {"TOX_PIP_VERSION": "18.1"},
+    },
+}
+
+PYTEST_PARAMETERS = sorted(itertools.product(TOX_VERSIONS, CASES))
+
+
+def setup_fresh_venv(tag, *extra_commands):
+    temp_dir = tempfile.TemporaryDirectory(prefix=tag)
+    venv_dir = os.path.join(temp_dir.name, 'venv')
+    subprocess.check_call(["virtualenv", venv_dir])
+    print('Created venv: %s' % venv_dir)
+    return temp_dir, venv_dir
+
+
+def install_deps(venv_dir, *deps):
+    """Install a dependency into the virtualenv"""
+    pip = os.path.join(venv_dir, 'bin', 'pip')
+    cmd = [pip, "install"] + list(deps)
+    subprocess.check_call(cmd, cwd=venv_dir, env={})
```

Comment:
note: `subprocess.run` doesn't exist in python 2.7. `subprocess.check_call` exists in 2.7 up through 3.7 (and raises an error on non-zero exit code, so no more assertions needed)